Maze.h

```diff
@@ -26,6 +26,8 @@ class Maze {
 
     bool isWall(NodeCoordinate pos);
     bool isWall(CellCoordinate pos, Direction dir);
+	bool isBorder(NodeCoordinate pos);
+	bool containsNode(NodeCoordinate pos);
```

Comment:
Not a huge fan of this name. It sounds like it should take a Node object.
How about `isValidCoordinate`